instant-xml/src/impls.rs

```diff
@@ -156,21 +125,37 @@ impl ToXml for &str {
     where
         W: fmt::Write,
     {
-        match serializer.consume_field_context() {
-            Some(field_context) => {
-                match field_context.attribute {
-                    Some(FieldAttribute::Attribute) => {
-                        serializer.add_attribute_value(self);
-                    }
-                    _ => {
-                        serializer.add_open_tag(&field_context)?;
-                        serializer.output.write_str(self)?;
-                        serializer.add_close_tag(field_context)?;
-                    }
-                }
-                Ok(())
-            }
-            None => Err(Error::UnexpectedValue),
-        }
+        let converted = special_xml_entities_handling(self);
+        DisplayToXml(&converted).serialize(serializer)
     }
 }
+
+impl ToXml for Cow<'_, str> {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        let converted = special_xml_entities_handling(self);
+        DisplayToXml(&converted).serialize(serializer)
+    }
+}
+
+fn special_xml_entities_handling(input: &str) -> String {
```

Comment:
Let's just call this `escape()`. It should return a `Cow<'_, str>` and should avoid writing into `result` in the case there are no escapeable characters.

I also think you're missing the `;` at the end of entities.

Let's not use `unsafe` for this for now, we can always add that later if we find that this is a bottleneck.

Reply:
done